source/bson-binary-vector/tests/README.md

```diff
@@ -50,7 +50,8 @@ MUST assert that the input float array is the same after encoding and decoding.
 
 #### To prove correct in an invalid case (`valid:false`), one MUST
 
-- raise an exception when attempting to encode a document from the numeric values, dtype, and padding.
+- when the vector field exists, raise an exception for encoding a document from the numeric values, dtype, and padding.
+- when the canonical_bson field exists, raise an exception for decoding it, as the field contains corrupted data that can't be decoded into a vector.
```

Comment:
It may be a breaking change for drivers to fail to read Vectors that have already been inserted without this requirement. 

How about drivers "fix" their serializers to stop making float32 vectors that have been created from the `canonical_bson` bytes?

Reply:
It should be fine to just "read" the Vector (as a BSON Binary). However, the specs need to clarify how to "decode/unserialize" to a sequence of float32 when the bytes for the vector elements are corrupted to store the sequence as the [description](https://github.com/mongodb/specifications/blob/master/source/bson-binary-vector/bson-binary-vector.md#data-structures) as well as expose the sequence to the driver users.

To prevent incompatibility issues, the exception doesn't have to be raised when reading the Binary of Vector from the server. It can be raised during the actual decoding/conversion (e.g. in the getter method of the float32 sequence).

Reply:
>  It can be raised during the actual decoding/conversion (e.g. in the getter method of the float32 sequence).

Perfect, and yeah that's how it works in node, our `toFloat32Array()` helper throws and `serialize`/`stringify` throw

Reply:
Will clarify the specs with an update.